### PR TITLE
Add sign-verify test

### PR DESCRIPTION
**Pull request by bouskaJ**
Adding the new test to verify sign-verify workflow. The test rely on pre-created keycloak user and signe the tagged image because SHAs are not supported by ttl.sh repository ATM but i think it's good enough for the sake of test.

### CODE REVIEW

**Comment by bouskaJ** [comment]
The test does not work on the gitHub actions CI because of the self-signed certificates on the local cluster (as discussed on the slack channel). @sallyom How would you approach it?

File: charts/trusted-artifact-signer/templates/tests/test-sign-verify.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      - name: buildah
+        image: quay.io/buildah/stable
+        command: ["/bin/sh", "-c"]
+        args:
+        - |
+            buildah pull alpine:latest
+            buildah tag alpine:latest ttl.sh/sigstore-test:5m
+            buildah push ttl.sh/sigstore-test:5m
+        securityContext:
+            privileged: true
```
Comment (Gregory-Pereira) [inline charts/trusted-artifact-signer/templates/tests/test-sign-verify.yaml:20]: Does this init container need to run as `privileged` just to pull, tag and push an image?

File: charts/trusted-artifact-signer/templates/tests/test-sign-verify.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          value: "https://keycloak-keycloak-system.$(OPENSHIFT_APPS_SUBDOMAIN)/auth/realms/$(OIDC_AUTHENTICATION_REALM)"
+        - name: REKOR_URL
+          value: "https://rekor.$(OPENSHIFT_APPS_SUBDOMAIN)"
+        - name: TUF_URL
+          value: "https://tuf.$(OPENSHIFT_APPS_SUBDOMAIN)"
+        command: ["/bin/sh", "-c"]
+        args:
+          - |
+            cosign initialize --mirror=$TUF_URL --root=$TUF_URL/root.json
+            TOKEN=$(curl -X POST -H "Content-Type: application/x-www-form-urlencoded" -d "username=jdoe" -d "password=secure" -d "grant_type=password" -d "scope=openid" -d "client_id=sigstore" $OIDC_ISSUER_URL/protocol/openid-connect/token |  sed -E 's/.*"access_token":"([^"]*).*/\1/')
```
Comment (Gregory-Pereira) [inline charts/trusted-artifact-signer/templates/tests/test-sign-verify.yaml:42]: Is this an issue to use hardcoded credentials like this? Im sure our routes / endpoints are available throughout the `securesign` repos in agregate, and couldn't someone sign code using this user against our stack as someone else?

**Review by Gregory-Pereira — COMMENTED** [review]
/lgtm but since this is not my area of expertise I had some questions

File: charts/trusted-artifact-signer/templates/tests/test-sign-verify.yaml (hunk 1 above)
Comment (bouskaJ) [reply charts/trusted-artifact-signer/templates/tests/test-sign-verify.yaml:20]: I am afraid that the container is not able to pull image without privileged context
```
sh-5.2# buildah pull alpine:latest
Error: error writing "0 0 4294967295\n" to /proc/17/uid_map: write /proc/17/uid_map: operation not permitted
ERRO[0000] writing "0 0 4294967295\n" to /proc/17/uid_map: write /proc/17/uid_map: operation not permitted 
ERRO[0000] (Unable to determine exit status)   
```

File: charts/trusted-artifact-signer/templates/tests/test-sign-verify.yaml (hunk 2 above)
Comment (bouskaJ) [reply charts/trusted-artifact-signer/templates/tests/test-sign-verify.yaml:42]: I'm not sure I understand your point exactly.
I followed the installation path described at https://github.com/securesign/sigstore-ocp/blob/main/quick-start-with-keycloak.md
As you can see there is a step to install keycloak resources from https://github.com/securesign/sigstore-ocp/tree/main/keycloak/resources. If you do, you end up with the user I used to sign the container in the test (see https://github.com/securesign/sigstore-ocp/blob/main/keycloak/resources/user.yaml#L14)

I know a pre-configured user can be seen as a pitfall, but I believe the keycloack scenario is worth covering.

**Comment by bouskaJ** [comment]
closing in favor of https://github.com/securesign/sigstore-ocp/pull/43

**Pull request closed without merging**